### PR TITLE
E S LINT!!!!

### PR DESCRIPTION
**Pull request by cloudyyoung**
Important rules:
- Imports order: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md
- Maximum lines (200 lines) and maximum statements (22 statements)
- Camel case variable names, always
- Arrow function declaration, always
- Use `===` instead of `==`, always

All these rules support auto-fixing.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "start": "vite --host --port 3000",
     "build": "vite build --out-dir build",
     "check-lint": "eslint . --ext .js,.jsx,.ts,.tsx --max-warnings 0 --ignore-path ../.gitignore && tsc --noEmit",
-    "lint": "eslint . --ext .js,.jsx,.ts,.tsx --fix",
+    "lint": "npm run check-lint --fix",
```
Comment (tim-macphail) [inline client/package.json:37]: At least on windows, the --fix option does nothing so `"npm run check-lint --fix"` == `"npm run check-lint"`

Comment (cloudyyoung) [reply client/package.json:37]: Oh... Reverted it back!

**Comment by tim-macphail** [comment]
Maybe let's wait until Bradley's PR is in, because it's likely to cause merge conflicts or issues passing checks if not. What do you think?

**Comment by tim-macphail** [comment]
hahahahahahaha might want to restart this one...

**Comment by cloudyyoung** [comment]
> hahahahahahaha might want to restart this one...

Ahhh :dizzy_face:

**Comment by cloudyyoung** [comment]
@tim-macphail Let's merge this now

**Review by tim-macphail — APPROVED** [review]

**Comment by tim-macphail** [comment]
@cloudyyoung my only concern is that it will make it harder for other devs to pass CI checks, so we'll have to make sure everyone's VSCode format on save is doing what it should be doing and all that with stuff in `.vscode`

**Comment by cloudyyoung** [comment]
> @cloudyyoung my only concern is that it will make it harder for other devs to pass CI checks, so we'll have to make sure everyone's VSCode format on save is doing what it should be doing and all that with stuff in `.vscode`

If all devs have all recommended plugins installed then we should be good

**Pull request merged**